### ISSUE TITLE
flux-perilog-run needs an option to run scripts serially (or to do it by default)

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#4044]
The `flux-perilog-run.py` script runs prolog/epilog scripts from a directory (`/etc/flux/system/{prolog,epilog}.d/*` by default) using the following iteration:

```python
scripts = [x for x in path.iterdir() if os.access(x, os.X_OK)]
```
However, `path.iterdir()` returns files in arbitrary order, so a specific order is not guaranteed. We should probably guarantee order by using `sorted(path.iterdir())`.

**Comment by grondo** [comment]
D'oh, I forgot that `flux-perilog-run.py` actually launches all matching scripts in parallel, so order is irrelevant.

https://github.com/flux-framework/flux-core/blob/80f0a4666b838b8f21b3101af647938c2310b97b/src/cmd/flux-perilog-run.py#L126-L129

I'm not sure at this point whether this was a good or bad design.

For now, in order to run a set of scripts in a specific order, a single script in the directory above could be used which does its own glob() (from a different directory).

Another idea would be an option to force the script to only run one epilog/prolog script at a time.

**Comment by grondo** [comment]
Changed the title to indicate that the real problem here is that prolog/epilog scripts may need to be run serially and we don't currently provide a way to do that.

**Comment by grondo** [comment]
The scripts in question here are the job-manager prolog/epilog, not the per-rank job prolog/epilog.
It is probably fine this way for now, so closing this issue.

We can open up a new issue in the future if this becomes a problem.
